src/pc/stats.rs

```diff
@@ -194,8 +197,31 @@ impl ParachainStats {
 	}
 
 	/// Update skipped slots count
-	pub fn on_skipped_slot(&mut self) {
+	pub fn on_skipped_slot(&mut self, update: &ParachainProgressUpdate) {
 		self.skipped_slots += 1;
+
+		let block = (update.block_number, update.block_hash);
+		if self.last_blocks_with_skipped_slots.len() < self.last_blocks_with_skipped_slots.capacity() {
+			self.last_blocks_with_skipped_slots.push(block)
+		} else {
+			let index = self.skipped_slots as usize % self.last_blocks_with_skipped_slots.capacity() - 1;
+			self.last_blocks_with_skipped_slots[index] = block;
+		}
+	}
```

Comment:
Why not use [VecDeque](https://doc.rust-lang.org/std/collections/struct.VecDeque.html) ? For each slot you just need to `push_back()` and `pop_front()` only when over max capacity.

Reply:
Would be nice if we also made a struct like `SkippedSlot { pub block_number: u32, pub block_hash: H256, }` instead of the tuple.